base/Enums.jl

```diff
@@ -47,19 +44,24 @@ macro enum(T,syms...)
             s = s.args[1]
             hasexpr = true
         else
-            throw(ArgumentError(string("invalid argument for Enum ", T, ", ", s)))
+            throw(ArgumentError(string("invalid argument for Enum ", typename, ", ", s)))
         end
-        if !isa(i, Integer) || !isbits(i)
-            throw(ArgumentError("Invalid value for Enum $T, $s=$i. Enum values must be integer bits types."))
+        if !isa(i, Integer)
+            throw(ArgumentError("Invalid value for Enum $typename, $s=$i. Enum values must be integers."))
         end
         if !Base.isidentifier(s)
-            throw(ArgumentError("Invalid name for Enum $T, $s is not a valid identifier."))
+            throw(ArgumentError("Invalid name for Enum $typename, $s is not a valid identifier."))
         end
         push!(vals, (s,i))
         I = typeof(i)
         enumT = length(vals) == 1 ? I : promote_type(enumT,I)
         lo = min(lo, i)
         hi = max(hi, i)
+        if !first
+            contiguous &= (i == prev+1)
```

Comment:
Another common idiom is Enum's which count down.  It would be nice to special case that as well.